### DISCUSSION TITLE
react-unity-webgl issue : A problem occurred while mounting abort({}) at ERROR

### DISCUSSION THREAD

**Discussion by chan27-2** [jeffreylanters/react-unity-webgl#207] [Help Wanted!]
### WebGL unity build is not working in the hosted site (AWS amplify) but working in **development mode**.
```
A problem occurred while mounting abort({}) at Error
    at jsStackTrace (https://....com/build/game/game.framework.js:2:16316)
    at stackTrace (https://...com/build/game/game.framework.js:2:16487)
    at abort (https://....com/build/game/game.framework.js:2:748)
    at https://....com/build/game/game.framework.js:2:25337
```
![Screenshot 2021-06-23 at 5 15 08 PM](https://user-images.githubusercontent.com/63861979/123091104-922aa680-d446-11eb-97a9-526f03defd50.jpg)

This is unity.ts:104 where its showing an error
![Screenshot 2021-06-23 at 6 15 23 PM](https://user-images.githubusercontent.com/63861979/123098647-fbaeb300-d44e-11eb-8bc4-f73c4dea2ad1.jpg)

in package.json
added `"react-unity-webgl": "^8.4.2"`

@jeffreylanters

**Comment by jeffreylanters** [comment]
Hi! Since the error is thrown by `game.framework.js`, I assume this is a problem with your Unity build. Does it run locally? Are you sure the path the build is correct when visiting via AWS? Could you send a screenshot of the network tab?

**Reply by chan27-2** [reply]
Yeah, it runs locally.
yes, I have given the path correctly.
![Screenshot 2021-06-26 at 1 33 39 AM](https://user-images.githubusercontent.com/63861979/123479429-8b568c00-d61e-11eb-8744-8e434566b356.jpg)

**Reply by davisg123** [reply]
+1 same issue, what Unity version are you using? I see it with 2020.3.12

**Reply by davisg123** [reply]
Digging more into the error, you can put a breakpoint at the `failed to asynchronously prepare wasm` log and I see this:

```
"CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0"
```

**Reply by chan27-2** [reply]
> +1 same issue, what Unity version are you using? I see it with 2020.3.12

yes I use the same version

**Reply by chan27-2** [reply]
But the thing is that it's not working only if I deploy it with **AWS amplify**,  in other hosting services when I deploy it it's perfect.

**Reply by invlabz** [reply]
Is there any solution for this? I am hosting using milesweb. Till yesterday it was running, but with the new build I am facing the same issue.

**Reply by invlabz** [reply]
> Is there any solution for this? I am hosting using milesweb. Till yesterday it was running, but with the new build I am facing the same issue.

Ok, in my case the browser cache is causing the error. On clearing the cache, working fine.

**Comment by daveyoung74** [comment]
In my case, I had to solve by adjusting my site's Content Security Template. The error message was misleading, but it seems like the site wasn't allowing the 'execution' of the script when hosted on S3.

**Comment by GrandArc** [comment]
> ### WebGL unity build is not working in the hosted site (AWS amplify) but working in **development mode**.
> ```
> A problem occurred while mounting abort({}) at Error
>     at jsStackTrace (https://....com/build/game/game.framework.js:2:16316)
>     at stackTrace (https://...com/build/game/game.framework.js:2:16487)
>     at abort (https://....com/build/game/game.framework.js:2:748)
>     at https://....com/build/game/game.framework.js:2:25337
> 
> in package.json added `"react-unity-webgl": "^8.4.2"`
> 
> @jeffreylanters

Did you manage to get this working? May I know what do you put on loaderUrl, dataUrl etc?
Since after I build my app under AWS Amplify, the unity loader is stuck at 90%. Thank you.

**Reply by chan27-2** [reply]
I have hosted it in AWS ec2 instead, amplify has some issues
